src/org/infinity/gui/converter/ConvertToMos.java

```diff
@@ -969,15 +1013,41 @@ private String getMosFileName(Path path)
     }
   }
 
-  private List<String> convert()
+  private ArrayList<String> getTableInputPaths() {
+    ArrayList<String> list = new ArrayList<>();
+    if(tfInputTableV1.getModel().getRowCount() > 0) {
+      for (int i = 0; i < tfInputTableV1.getModel().getRowCount(); i++) {
+        list.add(tfInputTableV1.getModel().getValueAt(i, 0).toString());
+      }
+
+      return list;
+    } else {
+      return null;
+    }
+  }
+
+  private ArrayList<String> getTableOutputNames() {
+    ArrayList<String> list = new ArrayList<>();
+    if(tfOutputTableV1.getModel().getRowCount() > 0) {
+      for (int i = 0; i < tfOutputTableV1.getModel().getRowCount(); i++) {
+        list.add(tfOutputTableV1.getModel().getValueAt(i, 0).toString());
+      }
```

Comment:
At least you can do that:
```suggestion
    final int count = tfOutputTableV1.getModel().getRowCount();
    if (count > 0) {
      final ArrayList<String> list = new ArrayList<>(count);
      for (int i = 0; i < count; i++) {
        list.add(tfOutputTableV1.getModel().getValueAt(i, 0).toString());
      }
```
But it is better to use your own `TableModel` class which will contain data you need